compiler/src/dmd/errors.d

```diff
@@ -529,6 +529,11 @@ private extern(C++) void verrorReport(const SourceLoc loc, const(char)* format,
             if (!global.gag)
             {
                 info.headerColor = Classification.warning;
+                if (global.params.v.messageStyle == MessageStyle.sarif)
+                {
+                    generateSarifReport(loc, format, ap, info.kind);
```

Comment:
Does this handle multiple errors yet?

Reply:
No, I was trying this yesterday and will do it again today. I modified three files to integrate full SARIF reporting into one. In `sarif.d`, errors were accumulated in a global array whenever a call was made from `errors.d` during error generation. At the end of compilation, `mars.d` called a function from `sarif.d` to generate and print the final report. But, this approach didn’t work as expected.  

The logic for accumulating and reporting errors seems correct, as I successfully implemented a similar approach in a standalone file, which works perfectly in isolation: [sarifmultiple](https://github.com/royalpinto007/d-drafts/blob/main/sarif_accumulator.d).  

Could you advise on what I might be missing for proper integration?  

Also, I have one question: Should the final report function be called at the end of `createModules` in `mars.d`, or is there a better place to hook into? Or should I create a PR first, and we can address the doubts there?